x-pack/plugins/fleet/server/routes/agent/handlers.ts

```diff
@@ -180,14 +183,25 @@ export const getAgentsHandler: RequestHandler<
   const esClient = coreContext.elasticsearch.client.asInternalUser;
   const esClientCurrentUser = coreContext.elasticsearch.client.asCurrentUser;
   const soClient = coreContext.savedObjects.client;
+  const { kuery } = request.query;
 
   try {
+    // validate kuery
+    const validationObj = validateKuery(kuery, [AGENTS_PREFIX], AGENT_MAPPINGS, true);
```

Comment:
Looking at `kbn-config-schema` library it seems that it allows to use a custom validation function: https://github.com/elastic/kibana/blob/main/packages/kbn-config-schema/README.md#custom-validation

I'll see if this can be used to call the `validateKuery` function rather than calling it inside the handler. Thanks for the suggestion!

Reply:
In https://github.com/elastic/kibana/pull/161064/commits/e52c163d2f5594871ffb01f85e651bb87b8a6093 I moved the validation function directly to the schema, I think it's more readable as well